CODE/_dev.py

```
@@ -1,33 +1,41 @@
+        elif key == "version":
+            config["System Settings"][key] = new_data
+        else:
+            log_dev.error(f"Invalid key: {key}")
+            return
+        with open(filename, "w", encoding="utf-8") as configfile:
             # noinspection PyTypeChecker
-            json.dump(data, f, indent=4)
-            f.truncate()
+            config.write(configfile, space_around_delimiters=False)
     except FileNotFoundError:
         log_dev.error(f"File not found: {filename}")
```

Ruff (0.8.0)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)